src/create_helpers.ts

```diff
@@ -507,9 +507,15 @@ export function createModule(doclet: INamespaceDoclet, nested: boolean, children
         flags |= ts.NodeFlags.NestedNamespace;
 
     if (children)
-        body = ts.createModuleBlock(children as ts.Statement[]);
-
-    const name = ts.createStringLiteral(doclet.name);
+				body = ts.createModuleBlock(children as ts.Statement[]);
+
+		const nameArray = [...doclet.name];
+		// Check if name is encapsulated in quotes
+		if (nameArray[0] === '"' && nameArray[nameArray.length] === '"' || nameArray[0] === "'" && nameArray[nameArray.length] === "'"){
+			nameArray.shift();
+			nameArray.pop();
+		};
+    const name = ts.createStringLiteral(nameArray.join(''));
```

Comment:
This should be length - 1 to check the last element.

Also you don't have to convert it to a string to check elements:

```suggestion
    let nameStr = doclet.name;
    if ((nameStr[0] === '"' && nameStr[name.length - 1] === '"')
        || (nameStr[0] === '\'' && nameStr[name.length - 1] === '\''))
    {
        nameStr = nameStr.substr(1, name.length - 2);
    }
    const name = ts.createStringLiteral(nameStr);
```